### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by ericaddison**
This is rebased off of #4, so that one should be finalized and submitted first...

- initial attempt at task 2 :)

### CODE REVIEW

**Comment by ericaddison** [comment]
Rebased off of #4 and cleaned up.

File: src/main/java/edu/texas/social_computing/hospitals/HRPP.java [hunk 1]
```diff
@@ -31,19 +31,28 @@ public static void run(HospitalTable hospitalTable, ResidentTable residentTable)
...
-        // Task 2
-        // if not matched
 
+            // Task 2
+            // if partners still not proximally matched
             // unmatch BOTH in the couple and add both back to the queue (now no one is dominant)
+            matching.unassign(ndResident);
+            matching.unassign(partner);
+
+            // ndResident should still be in queue, right???
```
Comment (p3ntak) [inline src/main/java/edu/texas/social_computing/hospitals/HRPP.java]: I believe that is correct. We'll have to note to make sure that is how @abruck1 does it. This refers to unmatched queue for reference.

File: src/main/java/edu/texas/social_computing/hospitals/HRPP.java [hunk 2]
```diff
@@ -31,19 +31,28 @@ public static void run(HospitalTable hospitalTable, ResidentTable residentTable)
...
 
             // reset non-dominant partner's preference list
+            ndResident.resetPreferences();
 
             // start from dominant partner's next hospital preference (changing the rank pointer)
+            residentTable.incrementResidentRankProgress(partner);
+            partner.setPrefsByProgress(residentTable.getResidentRankProgress(partner));
 
+            // run again
+            matching = HRP.run(hospitalTable, residentTable, unmatchedQueue);
```
Comment (p3ntak) [inline src/main/java/edu/texas/social_computing/hospitals/HRPP.java]: I think down here we will need to run the proximity violations check again and push any new people into violatingResidentsQ. Thoughts?

File: src/main/java/edu/texas/social_computing/hospitals/HRPP.java (hunk 1 above)
Comment (ericaddison) [reply src/main/java/edu/texas/social_computing/hospitals/HRPP.java]: Yes, I agree. Good call.

Comment (ericaddison) [reply src/main/java/edu/texas/social_computing/hospitals/HRPP.java]: Done.

File: src/main/java/edu/texas/social_computing/hospitals/HRPP.java (hunk 2 above)
Comment (ericaddison) [reply src/main/java/edu/texas/social_computing/hospitals/HRPP.java]: Agreed. Done.

**Pull request merged**